### PR TITLE
CCX-Notification-Service tests update

### PR DESCRIPTION
**Pull request by epapbak**
# Description

Additional tests for notification service for both destinations

Fixes [CCXDEV-7545](https://issues.redhat.com/browse/CCXDEV-7545)

## Type of change

- New test feature file
- New test scenario added into existing feature file

## Testing steps

run `WITHMOCK=1 make notification-service-tests` locally or within docker containers

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [x] updated documentation wherever necessary
* [x] new tests can be executed both locally and within docker container

### CODE REVIEW

File: features/ccx-notification-service/configuration.feature [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      And insights-content service is available on localhost:8082
+      And prometheus push gateway is available on localhost:9091
+
+
+  Scenario: Check that notification service exits with exit code 1 if no destination is configured
+    Given the service is expected to exit with code 1
+     When I start the CCX Notification Service with the --instant-reports command line flag
+          | val                                                   | var         |
+          | CCX_NOTIFICATION_SERVICE__KAFKA_BROKER__ENABLED       | false       |
+          | CCX_NOTIFICATION_SERVICE__SERVICE_LOG__ENABLED        | false        |
```
Comment (tisnik) [inline features/ccx-notification-service/configuration.feature:22]: will it work if not alligned?

File: features/ccx-notification-service/service_log.feature [hunk 2]
```diff
@@ -39,6 +39,49 @@ Feature: Service Log
           | CCX_NOTIFICATION_SERVICE__SERVICE_LOG__ENABLED  | true  |
      Then it should have sent 1 notification events to Service Log
       And the process should exit with status code set to 0
+     When I retrieve the service log events for the following clusters
+          | cluster name                         |
+          | 5d5892d4-2g85-4ccf-02bg-548dfc9767aa |
+     Then I should find the following log events for each cluster
+          | cluster name                         | num logs | service name
```
Comment (tisnik) [inline features/ccx-notification-service/service_log.feature:46]: nitpick - | at EOLNs?

File: features/ccx-notification-service/service_log.feature [hunk 3]
```diff
@@ -39,6 +39,49 @@ Feature: Service Log
...
+          | CCX_NOTIFICATION_SERVICE__KAFKA_BROKER__ENABLED    | false |
+          | CCX_NOTIFICATION_SERVICE__SERVICE_LOG__ENABLED     | true  |
+          | CCX_NOTIFICATION_SERVICE__SERVICE_LOG__CREATED_BY  | test  |
+     Then it should have sent 1 notification events to Service Log
+      And the process should exit with status code set to 0
+     When I retrieve the service log events for the following clusters
+          | cluster name                         |
+          | 5d5892d4-2g85-4ccf-02bg-548dfc9767aa |
+     Then I should find the following log events for each cluster
+          | cluster name                         | num logs | service name
```
Comment (tisnik) [inline features/ccx-notification-service/service_log.feature:66]: dtto

File: features/ccx-notification-service/service_log.feature [hunk 4]
```diff
@@ -39,6 +39,49 @@ Feature: Service Log
...
+  @rest-api
+  Scenario: Check that notification service does not send messages to service log if it cannot be rendered
+     When I insert 1 report with important total risk for the following clusters
+          | org id |  account number | cluster name                         |
+          | 1      |  1              | 5d5892d4-2g85-4ccf-02bg-548dfc9767aa |
+      And I start the CCX Notification Service with the --instant-reports command line flag
+          | val                                             | var   |
+          | CCX_NOTIFICATION_SERVICE__KAFKA_BROKER__ENABLED | false |
+          | CCX_NOTIFICATION_SERVICE__SERVICE_LOG__ENABLED  | true  |
+          | CCX_NOTIFICATION_SERVICE__DEPENDENCIES__TEMPLATE_RENDERER_SERVER  | unknown_url |
```
Comment (tisnik) [inline features/ccx-notification-service/service_log.feature:79]: allign? :)

**Review by tisnik — CHANGES_REQUESTED** [review]
LGTM in overall, just few nitpick

File: features/ccx-notification-service/configuration.feature (hunk 1 above)
Comment (epapbak) [reply features/ccx-notification-service/configuration.feature:22]: it works, but I actually prefer it to be aligned. Fixing :)

**Review by tisnik — APPROVED** [review]
ok

File: features/ccx-notification-service/service_log.feature (hunk 2 above)
Comment (epapbak) [reply features/ccx-notification-service/service_log.feature]: Added

**Review by Bee-lee — APPROVED** [review]
LGTM

**Pull request merged**